src/java/org/apache/cassandra/db/Mutation.java

```diff
@@ -663,8 +663,12 @@ public PartitionUpdateCollector(String keyspaceName, DecoratedKey key)
 
         public PartitionUpdateCollector add(PartitionUpdate partitionUpdate)
         {
-            assert partitionUpdate != null;
-            assert partitionUpdate.partitionKey().getPartitioner() == key.getPartitioner();
+            assert partitionUpdate != null : "Null updates are not allowed";
+            assert partitionUpdate.partitionKey().getPartitioner() == key.getPartitioner(): String.format("Update to key %s with partitioner %s (%s) had an update (%s) with a different partitioner! %s (%s)",
```

Comment:
this is failing often in CI and added this logging to give better details on what's going on.

```
java.lang.AssertionError
	org.apache.cassandra.db.Mutation$PartitionUpdateCollector.add(Mutation.java:667)
	org.apache.cassandra.db.commitlog.CommitLogReplayer$MutationInitiator$1.runMayThrow(CommitLogReplayer.java:308)
```


```
java.lang.RuntimeException: java.lang.AssertionError
	org.apache.cassandra.utils.Throwables.unchecked(Throwables.java:308)
	org.apache.cassandra.utils.Throwables.cleaned(Throwables.java:327)
	org.apache.cassandra.utils.FBUtilities.waitOnFuture(FBUtilities.java:544)
	org.apache.cassandra.db.commitlog.CommitLogReplayer.handleMutation(CommitLogReplayer.java:521)
	org.apache.cassandra.db.commitlog.CommitLogReader.readMutation(CommitLogReader.java:478)
```